test/CMakeLists.txt

```diff
@@ -145,23 +145,40 @@ endmacro()
 
 set(_regexp_dpcpp_backend_required "(xpu_api/algorithms|xpu_api/functional|xpu_api/numerics)")
 set(_regexp_switch_off_checked_it  "(test/general/header_order_ranges|test/parallel_api/algorithm/alg.sorting/alg.min.max|test/parallel_api/ranges|test/xpu_api/random/device_tests|test/xpu_api/random/interface_tests|test/xpu_api/random/statistics_tests|test/parallel_api/numeric/numeric.ops/transform_scan)")
+
+set(_regexp_test_pstl_offload_algorithms "(test/parallel_api/algorithm|test/parallel_api/memory|test/parallel_api/numeric|test/xpu_api)")
+set(_regexp_test_pstl_offload_headers "(test/pstl_offload)")
+set(_regexp_test_exclude_pstl_offload "(test/parallel_api/algorithm/alg.sorting/alg.binary.search|test/parallel_api/numeric/numeric.ops/.*_by_segment)")
+
 file(GLOB_RECURSE UNIT_TESTS "*.pass.cpp")
 foreach (_file IN LISTS UNIT_TESTS)
-
-    if (_file MATCHES "${_regexp_dpcpp_backend_required}")
-        if (ONEDPL_BACKEND MATCHES "^(dpcpp|dpcpp_only)$")
-            # Switch off checked iterators
+    # TODO: find a way to simplify this if
+    if (_file MATCHES "${_regexp_test_pstl_offload_headers}")
+        if (ONEDPL_BACKEND MATCHES "^(pstl_offload)$")
             onedpl_add_test(${_file} true)
-        else()
-            message(STATUS "Skip test ${_file} as it requires DPC++ backend")
         endif()
     else()
-        if (_file MATCHES "${_regexp_switch_off_checked_it}")
-            # Switch off checked iterators
-            onedpl_add_test(${_file} true)
+        if (ONEDPL_BACKEND MATCHES "^(pstl_offload)$")
+            if(_file MATCHES "${_regexp_test_pstl_offload_algorithms}" AND NOT _file MATCHES "${_regexp_test_exclude_pstl_offload}")
+                onedpl_add_test(${_file} true)
+            endif()
```

Comment:
Can this be a single conditional which checks for a match vs `_regexp_test_pstl_offload_algorithms` OR `_regexp_test_pstl_offload_headers` AND excludes `_regexp_test_exclude_pstl_offload`?

Reply:
This part was rewritten with only one if statement checking regular expressions.